src/main/java/com/zotov/edu/passportofficerestservice/controller/converter/PassportControllerDtoConverter.java

```diff
@@ -0,0 +1,43 @@
+package com.zotov.edu.passportofficerestservice.controller.converter;
+
+import com.zotov.edu.passportofficerestservice.controller.dto.response.PassportResponse;
+import com.zotov.edu.passportofficerestservice.controller.dto.response.PersonResponse;
+import com.zotov.edu.passportofficerestservice.repository.entity.Passport;
+import com.zotov.edu.passportofficerestservice.repository.entity.Person;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class PassportControllerDtoConverter {
+    public PassportResponse convertPassportToDto(Passport passport) {
+        return new PassportResponse(
+                passport.getNumber(),
+                passport.getGivenDate(),
+                passport.getDepartmentCode());
+    }
+
+    public List<PassportResponse> convertPassportListToDto(List<Passport> passport) {
+        return passport.stream().map(this::convertPassportToDto).collect(Collectors.toList());
+    }
+
+    public Page<PersonResponse> convertPersonToDto(Page<Person> personPage) {
+        List<PersonResponse> personResponses = convertPersonListToDto(personPage.getContent());
+        return new PageImpl<>(personResponses);
```

Comment:
Here we loose information about total number of elements in service, and current page.
Better to use `Page#map` instead

Reply:
The converter has been replaced by the map method of Page. Also, the page implementation for the Person repository has been fixed to not to miss the page size and page parameters.